cw/capture.py

```diff
@@ -1180,6 +1193,15 @@ def capture_otbn_vertical(ot, ktp, fw_bin, pll_frequency, capture_cfg, device_cf
             # Use the fixed seed.
             seed_used = seed_fixed
             expected_key = expected_fixed_key
+        elif capture_cfg["test_type"] == 'KEY':
+            # Use a random key with fixed redundancy.
+            ktp.key_len = key_bytes
+            random_key = ktp.next_key()
+            ktp.key_len = seed_bytes
+            intrandom = int.from_bytes(C, byteorder='little') + \
+                int.from_bytes(random_key, byteorder='little')
+            seed_used = intrandom.to_bytes(seed_bytes, byteorder='little')
```

Comment:
Similar to above, what about XOR with seed1?
Also, the mask (seed2) is still generated independently from the key. Wasn't the initial idea to calculate seed2 (the mask) in dependency of the fixed/random key and a random seed1, so that we can see leakage of the key and not the seed?

Reply:
Thanks @wettermo for looking into this.
It is probably confusing that the terminology in the discussion document is not consistent with the variable names. In this code I was trying to be consistent with the variable names that we already use, rather than changing everything to fit the discussion document. Here is how the two terminologies relate (bold font for **discussion document**):
C                   **C**
key_fixed      **fixed_key**
random_key  **random_key**
intseed          **C + fixed_key**
intrandom      **C + random key**
seed_used    **C + random_key** or **C + fixed_key**
mask             **seed1**
computed by ibex **seed2**

I believe that it's functionally correct, but please double check.
I could probably change some variable names to make the code more intuitive.

Reply:
Thanks for pointing this out, probably makes it easier to understand.
But I am still uncertain if this is how we/you intended this. Because seed2 = mask is not computed by Ibex. Ibex just hands over this value to the OTBN as second share. This means that this value is still independent of the fixed/random key.

Reply:
We should streamline things here and link the discussion document to the PR and/or even to the commit.
If there is already confusion in the PR, things might get worse later on, when we eventually have a second look on the code.

Reply:
Thanks for pointing this out, might clear up things a bit. 
But still, I thought the mask (and share2) is equal to seed2, while seed1 is equal to the share1 (the seed).

Reply:
Sorry, I was referring to the binary generated from [OT #17502](https://github.com/lowRISC/opentitan/pull/17502) which hasn't yet been merged. Currently the XOR on ibex is happening in function `p256_run_keygen()`